sonar-scala-plugin/src/main/java/org/sonarsource/scala/externalreport/scalastyle/ScalastyleFamilySensor.java

```diff
@@ -0,0 +1,148 @@
+import org.sonar.api.batch.sensor.issue.NewExternalIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.analyzer.commons.ExternalReportProvider;
+import org.sonarsource.analyzer.commons.ExternalRuleLoader;
+import org.sonarsource.scala.plugin.ScalaPlugin;
+
+public class ScalastyleFamilySensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(ScalastyleFamilySensor.class);
+
+  static final String SCALASTYLE_LINTER_KEY = "scalastyle";
+  static final String SCALASTYLE_LINTER_NAME = "Scalastyle";
+
+  static final String SCAPEGOAT_LINTER_KEY = "scapegoat";
+  static final String SCAPEGOAT_LINTER_NAME = "Scapegoat";
+
+  private static final String SCALASTYLE_PACKAGE_PREFIX = "org.scalastyle.";
+  private static final String SCAPEGOAT_PACKAGE_PREFIX = "com.sksamuel.scapegoat.inspections.";
+
+  public static final String REPORT_PROPERTY_KEY = "sonar.scala.scalastyle.reportPaths";
+
+  private static final int MAX_LOGGED_FILE_NAMES = 20;
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor
+      .onlyWhenConfiguration(conf -> conf.hasKey(REPORT_PROPERTY_KEY))
+      .onlyOnLanguage(ScalaPlugin.SCALA_LANGUAGE_KEY)
+      .name("Import of Scalastyle/Scapegoat issues");
+  }
+
+  @Override
+  public void execute(SensorContext context) {
+    Set<String> unresolvedInputFile = new HashSet<>();
+    List<File> reportFiles = ExternalReportProvider.getReportFiles(context, REPORT_PROPERTY_KEY);
+    reportFiles.forEach(report -> importReport(report, context, unresolvedInputFile));
+    logUnresolvedInputFiles(unresolvedInputFile);
+  }
+
+  private static void logUnresolvedInputFiles(Set<String> unresolvedInputFile) {
+    if (unresolvedInputFile.isEmpty()) {
+      return;
+    }
+    String fileList = unresolvedInputFile.stream().sorted().limit(MAX_LOGGED_FILE_NAMES).collect(Collectors.joining(";"));
+    if (unresolvedInputFile.size() > MAX_LOGGED_FILE_NAMES) {
+      fileList += ";...";
+    }
+    LOG.warn("Fail to resolve {} file(s). No issues imported related to file(s): {}", unresolvedInputFile.size(), fileList);
+  }
+
+  private static void importReport(File reportPath, SensorContext context, Set<String> unresolvedInputFile) {
+    try (InputStream in = new FileInputStream(reportPath)) {
+      LOG.info("Importing {}", reportPath);
+      ScalastyleXmlReportReader.read(in, (file, line, source, message) -> saveIssue(context, file, line, source, message, unresolvedInputFile));
+    } catch (IOException | XMLStreamException | RuntimeException e) {
+      LOG.error("No issues information will be saved as the report file '{}' can't be read.", reportPath, e);
+    }
+  }
+
+  private static void saveIssue(SensorContext context, String file, String line, String source, String message, Set<String> unresolvedInputFile) {
```

Comment:
What is "source"?

Reply:
It's the rule key, but because it's called "source" in the scalastyle report, I keep this naming.